components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml

```diff
@@ -2005,6 +2005,25 @@ document.observe('xwiki:dom:loaded', function() {
       new PhenoTips.widgets.ChartGenerator($('charts'), chartDataSources);
     });
 
+    //  Create a warning when measurements are being entered but no birth date is specified
+    var dateOfBirth = $(document.documentElement).down('input[name$=date_of_birth]');
+    //var warningInsertionPoint = $$('.measurement-info &gt; .expand-tools')[0] ? $$('.measurement-info &gt; .expand-tools')[0] : $('HMeasurements');
+    if (dateOfBirth){
+        _updateNoBirthDateWarning();
+        dateOfBirth.observe('blur', _updateNoBirthDateWarning);
```

Comment:
You should also listen for `xwiki:date:changed`. Actually listening only for that event should be enough.